osu.Game/Graphics/UserInterface/RollingCounter.cs

```diff
@@ -46,57 +47,71 @@ public abstract class RollingCounter<T> : Container, IHasAccentColour
         public virtual T DisplayedCount
         {
             get => displayedCount;
-
             set
             {
                 if (EqualityComparer<T>.Default.Equals(displayedCount, value))
                     return;
 
                 displayedCount = value;
-                DisplayedCountSpriteText.Text = FormatCount(value);
+                if (displayedCountSpriteText != null)
+                    displayedCountSpriteText.Text = FormatCount(value);
             }
         }
 
         public abstract void Increment(T amount);
 
+        private float textSize = 40f;
+
         public float TextSize
         {
-            get => DisplayedCountSpriteText.Font.Size;
-            set => DisplayedCountSpriteText.Font = DisplayedCountSpriteText.Font.With(size: value);
+            get => displayedCountSpriteText?.Font.Size ?? textSize;
+            set
+            {
+                if (TextSize == value)
+                    return;
+
+                textSize = value;
+                if (displayedCountSpriteText != null)
+                    displayedCountSpriteText.Font = displayedCountSpriteText.Font.With(size: value);
+            }
         }
```

Comment:
I sorta get why this is necessary (`DisplayedCountSpriteText` is no longer guaranteed to be initialised at call time) but this is almost too ugly to let live in its current state IMO - this looks like it's not encapsulated well enough

maybe something like a `SpriteTextFactory` (a dumb object that can create a sprite text on demand) is something we might want here - unsure whether that's too enterprise™ or if it fits without seeing what the actual use case is for the legacy one properly